### DISCUSSION TITLE
Pip-based rez installation detected warning with installation script

### DISCUSSION THREAD

**Discussion by OlivierArgentieri** [AcademySoftwareFoundation/rez#1296] [Q&A] [answered]
Environment:
OS : Windows 10
rez: Latest version (2.109.0)
Problem:
The warning "Pip-based rez installation detected" appear even when rez is not installed with pip.

To reproduce :
Install Python 3.7 win the .exe file
Then launch installation of REZ with `python ./rez-2.109.0/install.py ./__install__`.

After some research i think i have found something.
The function: check_production_install in `__install__\Lib\site-packages\rez\cli\_entry_points.py`, check if the file `.rez_production_install` exist in `__install__\Scripts` but the file exist in `__install__\Scripts\rez`.

Maybe i have made something wrong ?
Thank you

**Comment by instinct-vfx** [comment, marked as answer]
The path you need to add and the executables you need to use are the ones in `\Scripts\rez`. The installer should remove the ones direlctly in `Scripts` but there is a bug preventing this (see PR https://github.com/nerdvegas/rez/pull/1259).

The installer script patches the executables and moves them to the subfolder in order to isolate the rez execution environment from the environments it creates. This is not possible with a pip install, hence the warning. You should not see the warning if you add `Scripts\rez` to `%PATH%`.

If you look at the check (https://github.com/nerdvegas/rez/blob/1ecc85f638d11d70ed78d4bd9c5cdc6f32ac58c4/src/rez/cli/_entry_points.py#L52) it checks for the file in the same location as the executable. So if you run the executable from `Scripts\rez` it will find the file and the error will go away.

**Reply by OlivierArgentieri** [reply]
Thank you for your help !
